sonar-java-plugin/src/test/java/org/sonar/plugins/java/JavaPluginTest.java

```diff
@@ -66,4 +69,29 @@ public void sonarqube_7_2_extensions() {
     assertThat(context.getExtensions()).hasSize(34);
   }
 
+  @Test
+  public void sonarqube_7_4_extensions() {
+    SonarRuntime runtime = SonarRuntimeImpl.forSonarQube(VERSION_7_4, SonarQubeSide.SERVER);
+    Plugin.Context context = new Plugin.Context(runtime);
+    javaPlugin.define(context);
+    assertThat(context.getExtensions()).hasSize(35);
+  }
+
+  @Test
+  public void use_AnalysisWarningsWrapper_before_SQ_7_4_and_sonarlint() {
+    SonarRuntime runtime = SonarRuntimeImpl.forSonarQube(Version.create(7, 3), SonarQubeSide.SCANNER);
+    Plugin.Context context = new Plugin.Context(runtime);
+    javaPlugin.define(context);
+    assertThat(context.getExtensions()).doesNotContain(AnalysisWarningsWrapper.class);
+
+    runtime = SonarRuntimeImpl.forSonarQube(ANALYSIS_WARNINGS_MIN_SUPPORTED_SQ_VERSION, SonarQubeSide.SCANNER);
```

Comment:
I don't think for testing we should use this constant but rather configure the version using VERSION_7_4 and make the constant ANALYSIS_...  a private static final variable in java plugin to better encapsulate things.